### PR TITLE
Add project repository url

### LINKED ISSUE #634: We need another url field
A project currently has a 'project url' field 'for this project\s [sic] home page'

I think there should be another field and they should be
- project website
- project code repository

### PR DESCRIPTION
**Pull request by sonlinux**
Fix #634 

Added project repository url to the project model, form and project detail page above the project url.

inasafe

### CODE REVIEW

File: django_project/base/migrations/0019_project_project_repository_url.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+    dependencies = [
+        ('base', '0018_auto_20180220_0339'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='project',
+            name='project_repository_url',
+            field=models.URLField(help_text='A repository url for this project i.e github repository url', null=True, blank=True),
```
Comment (timlinux) [inline django_project/base/migrations/0019_project_project_repository_url.py]: GitHub

Comment (timlinux) [reply django_project/base/migrations/0019_project_project_repository_url.py]: e.g. not i.e.

File: django_project/base/models/project.py [hunk 2]
```diff
@@ -158,6 +158,13 @@ class Project(models.Model):
         null=True
     )
 
+    project_repository_url = models.URLField(
+        help_text=u'A repository url for this project '
+                  u'i.e github repository url',
```
Comment (timlinux) [inline django_project/base/models/project.py]: e.g. GitHub

Also please always use complete sentances (with full stop terminating them)

Comment (sonlinux) [reply django_project/base/models/project.py]: Alright @timlinux thanks
Updated.

File: django_project/base/migrations/0019_project_project_repository_url.py (hunk 1 above)
Comment (sonlinux) [reply django_project/base/migrations/0019_project_project_repository_url.py]: Alright @timlinux thanks
Updated.

Comment (timlinux) [reply django_project/base/migrations/0019_project_project_repository_url.py]: You still have the old text....

```field=models.URLField(help_text='A repository url for this project i.e github repository url', null=True, blank=True),```

becomes

```
field=models.URLField(help_text='A repository url for this project e.g. GitHub repository url.', null=True, blank=True),
```

File: django_project/base/models/project.py [hunk 3]
```diff
@@ -158,6 +158,13 @@ class Project(models.Model):
         null=True
     )
 
+    project_repository_url = models.URLField(
+        help_text=u'A repository url for this project '
+                  u'e.g a github repository url.',
```
Comment (timlinux) [inline django_project/base/models/project.py]: Please check carefully my comment:

```
      help_text=u'A repository url for this project '
                 u'e.g a github repository url.',
```

should be

```
      help_text=u'A repository URL for this project '
                 u'e.g, a GitHub repository URL.',
```

Also your string contruction is strange, it should be:

```
      help_text=(
          u'A repository url for this project '
          'e.g. a GitHub repository URL.'),
```

**Comment by timlinux** [comment]
@sonlinux from our discussion this would come into code by proxy class - since I am now resorting to giving you exactly the code you needed to write.

**Comment by sonlinux** [comment]
Ah sorry about that @timlinux 
I corrected then reset the commit because I wanted to just add 'for example' but I commited before that change.
Ill pay more attention.
I have updated

**Comment by timlinux** [comment]
Not all the code is updated - also use URL rather than url in your text.

**Comment by Gustry** [comment]
Maybe you should display the link somewhere in a template?

For the migration, remove the old migration file you created in this PR and re-run the migration command. Or just re-run the make update migration command to add another migration file

File: django_project/base/models/project.py [hunk 4]
```diff
@@ -159,8 +159,8 @@ class Project(models.Model):
     )
 
     project_repository_url = models.URLField(
-        help_text=u'A repository URL for this project '
-                  u'e.g, a Github repository URL.',
+        help_text=u'A repository URL for this project. '
+                  u'For instance a path to the project\s Github repository.',
```
Comment (Gustry) [inline django_project/base/models/project.py]: As Tim mentioned, better to put bracket when you are doing multiline strings.
```
help_text=(
    u'Looooong string'),
```
Example: https://github.com/inasafe/inasafe/blob/develop/safe/definitions/layer_purposes.py#L86

Also `GitHub` not `github`: https://en.wikipedia.org/wiki/GitHub 
And also the apostrophe after project?

**Comment by sonlinux** [comment]
Thanks @Gustry 
I have displayed the link above project home url with an icon.
And for the migrations I have rerun and they are reflecting.

File: django_project/base/models/project.py [hunk 5]
```diff
@@ -158,6 +158,13 @@ class Project(models.Model):
         null=True
     )
 
+    project_repository_url = models.URLField(
+        help_text=u'A repository URL for this project. '
+                  u'For instance a path to the project\'s Github repository.',
```
Comment (timlinux) [inline django_project/base/models/project.py]: Github -> GitHub

Comment (sonlinux) [reply django_project/base/models/project.py]: :-) Okay @timlinux 
Thanks

File: django_project/base/migrations/0019_project_project_repository_url.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+    dependencies = [
+        ('base', '0018_auto_20180220_0339'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='project',
+            name='project_repository_url',
+            field=models.URLField(help_text="A repository URL for this project. For instance a path to the project's GitHub repository.", null=True, blank=True),
```
Comment (timlinux) [inline django_project/base/migrations/0019_project_project_repository_url.py]: PEP8

**Comment by timlinux** [comment]
LGTM thanks

File: django_project/base/migrations/0019_project_project_repository_url.py (hunk 6 above)
Comment (Gustry) [reply django_project/base/migrations/0019_project_project_repository_url.py]: This is a migration file. They are autogenerated by django so excluded from the PEP8 test in Travis and makefile.

**Pull request merged**